src/SmoFacade/User.cs

```diff
@@ -1,18 +1,62 @@
 ﻿namespace AgDatabaseMove.SmoFacade
 {
+  using System.Collections.Generic;
+  using Smo = Microsoft.SqlServer.Management.Smo;
+
+  public class UserProperties
+  {
+    public string Name { get; set; }
+    public IEnumerable<RoleProperties> Roles { get; set; }
+    public Smo.DatabasePermissionSet Permissions { get; set; }
+    public string LoginName { get; set; }
+  }
+
   public class User
   {
     private readonly Server _server;
-    private readonly Microsoft.SqlServer.Management.Smo.User _user;
+    private readonly Database _database;
+    private Smo.User _user;
 
-    public User(Microsoft.SqlServer.Management.Smo.User user, Server server)
+    public User(Smo.User user, Server server, Database database)
     {
       _user = user;
       _server = server;
+      _database = database;
+    }
+
+    public User(UserProperties userProperties, Server server, Database database)
+    {
+      _server = server;
+      _database = database;
+      ConstructUser(userProperties);
     }
 
     public string Name => _user.Name;
 
+    private void ConstructUser(UserProperties userProperties)
+    {
+      _user = new Smo.User(_database._database, userProperties.Name) { Login = userProperties.LoginName };
+      _user.Create();
+
+      foreach (var role in userProperties.Roles) AddRole(role);
+      GrantPermission(userProperties.Permissions);
+    }
+
+    public void AddRole(RoleProperties role)
+    {
+      _user.AddToRole(role.Name);
+    }
+
+    public void GrantPermission(Smo.DatabasePermissionSet permissions)
+    {
+      _database._database.Grant(permissions, Name);
+    }
+
+    public void Drop()
+    {
+      _user.Drop();
+    }

```

Comment:
I'm not sure what this does... Does calling `user.drop` remove it from the database? (ie. does this user object have a server/database property tied to it?)

Reply:
From my understanding, the user object has a database property tied to it which has a server property tied to it, yes.
I believe the drop is performed specifically from a non-query statement execution, i.e. `DROP USER _username`_

Reply:
Can you please test and confirm this?  "understanding" from documentation is far from working IRL

Reply:
I've tested it -- it successfully drops the user from the database.